text/0060-data-memos.md

```diff
@@ -0,0 +1,111 @@
+
+## 0x0300 Custom Memo
+
+The 64-byte memo data is laid out as follows:
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 64     | Custom Data           |
+
+This memo has no defined structure or data boundaries. All bytes are left for the client to fill with arbitrary data. Developers can create their own memo layouts for their applications using this memo type.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There are some risks associated with allowing clients full control of the ememo field. Inevitably, applications will introduce their own memo types and/or protocols using custom memo types. We cannot guarantee that end users won't use an account on more than one application (nor that developers will discourage this practice). The dangers of using custom memo types must therefore be documented clearly to try and avoid this as much as possible.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Sender and Destination With Data Memos
+
+The current RTH memos are unnecessarily limiting. Nearly half of all sender and destination memo types is unused space. Opening this space up completely to clients will allow them to come up with their own uses for the ememo field. At the same time, the Sender and Destination with Data Memos allow for clients to use RTH.
+
+It may also be possible to modify the existing sender and destination memos to allow for the 32 unused bytes to be written to. Rather than modifying the old memo types, new ones are proposed so that the memo type can signify to applications whether or not the data stored in those 32 bytes is meaningful.
+
+## Custom Memo
+
+For clients that do not need RTH, it wouldn't make sense for them to use the Sender and Destination With Data Memos. Providing this memo type gives clients complete freedom to implement whatever features they want using ememos.
+
+It could also be possible to allow clients to use the memo type bytes as well, but this proposal does not allow for that. The Custom Memo was given a fixed type. Though it generally won't be supported, if a user shares one account across multiple different clients, developers have a standard way of recognizing unsupported data and choosing to ignore (rather than try to process) invalid memos for their application.
+
+There could also be a separate memo type for the payload and change `TxOut`s, but this may be more complicated than necessary. Clients do not need to rely on checking memo types to differentiate change `TxOut`s.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Would it be beneficial to have two different Custom Memo types: one for the payload `TxOut` and one for the change `TxOut`? Maybe the memo builder can accept a 'data' field and a 'change data' field. If the 'change data' field is not set, then the 'data' field is used for both outputs.
```

Comment:
The more I think about this, the more I think it is a good idea. But should the type be 0x0400 or 0x0301?